File: src/main/java/nl/martijnklene/api/infrastructure/ApiApplication.java
package nl.martijnklene.api.infrastructure;

import org.axonframework.spring.config.EnableAxon;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.ComponentScans;

@ComponentScans(
    @ComponentScan(value = "nl.martijnklene.api")
)
@EntityScan(basePackages  = "nl.martijnklene.api")
@SpringBootApplication
public class ApiApplication {

	public static void main(String[] args) {
		SpringApplication.run(ApiApplication.class, args);
	}
}

File: src/main/kotlin/nl/martijnklene/api/infrastructure/ApiApplication.kt
package nl.martijnklene.api.infrastructure

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.PropertySource
import org.springframework.context.annotation.PropertySources

@SpringBootApplication
@PropertySources(
    PropertySource("classpath:application.properties"),
    PropertySource("file:/run/secrets/blog_api_config", ignoreResourceNotFound = true)
)
class ApiApplication

fun main(args: Array<String>) {
    runApplication<ApiApplication>(*args)
}
